Builder/ElasticSearchBuilder.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+namespace Markup\NeedleBundle\Builder;
+
+use Markup\NeedleBundle\Filter\Field\ArrayFilterField;
+use Markup\NeedleBundle\Filter\Field\ExistsFilterField;
+use Markup\NeedleBundle\Filter\Field\RangeFilterField;
+use Markup\NeedleBundle\Filter\Field\TextFilterField;
+
+class ElasticSearchBuilder implements ElasticDocumentInterface
```

Comment:
As a concept I think this shouldn't be called `Builder`.. its also using `arrays` so much its hard to see value in it, its not really a like for like with the `SolariumSelectQueryBuilder` so Im having to hard time understanding why it exists in Needle.. if Needle is at least meant to provide functionality across both platforms.